### PR TITLE
Fix invalid identity in blueprint.py

### PR DESCRIPTION
**Pull request by spauka**
Fixes several erroneous identity comparisons that are showing up as warnings in Python 3.10 on import. Specifically when comparing `int` and empty strings, `!=` should be used instead of `is not`.

In addition, .gitignore is extended to include extra files/directories from https://github.com/github/gitignore/blob/main/Python.gitignore to support development installs in python. 

@jenshnielsen

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,152 @@
...
+
+# Cython debug symbols
+cython_debug/
+
+# PyCharm
+#  JetBrains specific template is maintainted in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+#.idea/
```
Comment (jenshnielsen) [inline .gitignore:152]: idealy this should end in a new line

**Review by jenshnielsen — APPROVED** [review]
Looks good to me. @WilliamHPNielsen @Dominik-Vogel any comments?

File: .gitignore (hunk 1 above)
Comment (spauka) [reply .gitignore]: Absolutely right :)

**Review by Dominik-Vogel — APPROVED** [review]

File: broadbean/blueprint.py [hunk 2]
```diff
@@ -878,7 +872,7 @@ def _subelementBuilder(blueprint: BluePrint, SR: int,
             chunk = int(np.round(dur*SR))
             marker[ind:ind+chunk] = 1
 
-    output = np.array(output)  # TODO: Why is this sometimes needed?
```
Comment (jenshnielsen) [inline broadbean/blueprint.py:881]: should we just remove that todo. It seems fairly obvious what is going on. Above output is defined as a list be we want to return an np array

Comment (spauka) [reply broadbean/blueprint.py:881]: Agreed with this as well :)

Take a look at this change - we now create the array with `np.fromiter` with preallocation which should be slightly faster than what it previously was.

**Review by jenshnielsen — APPROVED** [review]

**Pull request merged**